src/sif/autocost.cc

```diff
@@ -412,7 +412,8 @@ bool AutoCost::Allowed(const baldr::DirectedEdge* edge,
       ((pred.restrictions() & (1 << edge->localedgeidx())) && !ignore_restrictions_) ||
       edge->surface() == Surface::kImpassable || IsUserAvoidEdge(edgeid) ||
       (!allow_destination_only_ && !pred.destonly() && edge->destonly()) ||
-      (pred.closure_pruning() && IsClosed(edge, tile))) {
+      (pred.closure_pruning() && IsClosed(edge, tile)) ||
+      (!use_unpaved_roads_ && !pred.unpaved() && edge->unpaved())) {
```

Comment:
i like this trick of using a bit to specify whether the predecessor was or wasnt unpaved instead of keeping track of the pruning state. i think its easier to reason about than setting and unsetting the pruning. we should switch to doing this for closures. cc @mandeepsandhu